Utils/HandlerUtil.py

```diff
@@ -61,26 +61,62 @@
 import base64
 import json
 import time
+
+from xml.etree import ElementTree
 from os.path import join
 from Utils.WAAgentUtil import waagent
 from waagent import LoggerInit
 
 DateTimeFormat = "%Y-%m-%dT%H:%M:%SZ"
 
+MANIFEST_XML = "manifest.xml"
+
 class HandlerContext:
     def __init__(self,name):
         self._name = name
         self._version = '0.0'
         return
 
 class HandlerUtility:
-    def __init__(self, log, error, short_name):
+    def __init__(self, log, error, s_name=None, l_name=None, version=None):
```

Comment:
I'm not very happy with my solution.  I would prefer that the code just know the correct values for long name, short name, and version, but I'm allowing it to be overridden.  (Actually, I would prefer a lot of things about the code, but ...) The user of of *... is None or ... is None or ... is None* is goofy.  The only caller of this is a test that I question the value of.  I'm thinking about making the parameters list just (self, log, error), and dropping the rest.

Thoughts?